### DISCUSSION TITLE
Generic GIT release tagging

### DISCUSSION THREAD

**Discussion by mszalbach** [jreleaser/jreleaser#1217] [General]
Hi,

Did I understand it correctly that the generic releaser did not tag and publish a tag on release?
Since I got this output from testing against Bitbucket where I can use a http or ssh access with mvn jreleaser:full-release:

        [INFO] Storing changelog: target\jreleaser\release\CHANGELOG.md
        [INFO] Calculating checksums
        [INFO]   [checksum] No files configured for checksum. Skipping
        [INFO] Signing files
        [INFO]   [sign] Signing is not enabled. Skipping
        [INFO] Deploying is not enabled. Skipping
        [INFO] Uploading is not enabled. Skipping
        [INFO] Releasing to a generic Git repository is not supported
        [INFO] Announcing release
        [INFO] Announcing is not enabled. Skipping
       
Thanks and regards,

Marcel

**Comment by aalmiray** [comment]
As far as I recall the generic git released will create a local tag and *should* push said tag to the remote git repository.
However, I can see in the log that there's a missing `Tagging x.y.z` message right after the warning stating that releasing for generic is not supported. This might be a bug. Do you have access to the full `trace.log`? It should be inside `target\jreleaser\trace.log`.

**Reply by aalmiray** [reply]
FWIW I was able to tag a remote BitBucket repository [https://bitbucket.org/aalmiray_bb/app](https://bitbucket.org/aalmiray_bb/app) using an app password as the value for `JRELEASER_GENERIC_TOKEN`:

```
$ jreleaser release 
[INFO]  JReleaser 1.4.0
[INFO]  Configuring with jreleaser.yml
[INFO]    - basedir set to /private/tmp/app
[INFO]    - outputdir set to /private/tmp/app/out/jreleaser
[INFO]  Reading configuration
[INFO]  git-root-search set to false
[INFO]  Loading variables from /Users/aalmiray/.jreleaser/config.toml
[INFO]  Validating configuration
[INFO]  Strict mode set to false
[INFO]  Project version set to 1.0.0
[INFO]  Release is not snapshot
[INFO]  Timestamp is 2023-02-22T17:25:45.206454+01:00
[INFO]  HEAD is at 3150708
[INFO]  Platform is osx-x86_64
[INFO]  dry-run set to false
[INFO]  Generating changelog
[INFO]  Storing changelog: out/jreleaser/release/CHANGELOG.md
[INFO]  Calculating checksums
[INFO]    [checksum] target/distributions/app/app-1.0.0.zip.sha256
[INFO]    [checksum] target/distributions/app/app-1.0.0.tar.gz.sha256
[INFO]  Signing files
[INFO]    [sign] Signing is not enabled. Skipping
[INFO]  Deploying is not enabled. Skipping
[INFO]  Uploading is not enabled. Skipping
[INFO]  Releasing to a generic Git repository is not supported
[INFO]  Tagging https://aalmiray_bb@bitbucket.org/aalmiray_bb/app.git
[INFO]  pushing to https://aalmiray_bb@bitbucket.org/aalmiray_bb/app.git
[INFO]  Writing output properties to out/jreleaser/output.properties
[INFO]  JReleaser succeeded after 1.233 s
```

The release configuration looks like this

```yml
release:
  generic:
    overwrite: true
    changelog:
      formatted: ALWAYS
      format: '- {{commitShortHash}} {{commitTitle}}'
      contributors:
        format: '- {{contributorName}}{{#contributorUsernameAsLink}} ({{.}}){{/contributorUsernameAsLink}}'
```

**Comment by mszalbach** [comment]
Thanks a lot for the fast reply. I did a short test with the standalone JReleaser and I can see that the trace did not say skipTag true like the maven plugin does and it is trying to push something.
So probally something in my pom.xml interfering with some default settings.

I will do some more tests tommorow. I guess I was too blinded by the "Releasing to a generic Git repository is not supported".

**Reply by aalmiray** [reply]
I had the same issue with the standalone configuration because the release section explicitly defined a name for the repository that did not match the actual name. Had `foobar` instead of `app`. I'd recommend not setting values for `owner`, `name`, and `username` so that defaults kick in.

**Reply by mszalbach** [reply]
Thanks that explains it. I have first used owner to correct the username to match my issued token. 
For the standalone version I needed to configure username to be able to authenticate, since the owner of the token and the Bitbucket project name are different.

**Comment by elopezvalecky** [comment]
Does jreleaser support **SSH**? currently doing local testing and I get this error

> org.eclipse.jgit.api.errors.TransportException: ssh://git@\*\*\*/\*\*\*/\*\*\*.git: remote hung up unexpectedly

I take a look around the source code and I don see any reference that is supported

**Reply by aalmiray** [reply]
I believe JReleaser expects `https` at the moment. Would have to investigate the consequences of enabling `ssh`.
